core-bundle/src/Resources/contao/library/Contao/Template.php

```diff
@@ -399,6 +399,46 @@ public function trans($strId, array $arrParams=array(), $strDomain='contao_defau
 		return System::getContainer()->get('translator')->trans($strId, $arrParams, $strDomain);
 	}
 
+	/**
+	 * Helper method to allow quick access in the Contao templates for raw output.
+	 * Be careful when using this. It must NOT be used within regular HTML when $value
+	 * is uncontrolled user input. It's useful to ensure raw values within e.g. <code> examples
+	 * or JSON-LD arrays.
+	 */
+	public function raw(string $value): string
+	{
+		return StringUtil::revertInputEncoding($value);
```

Comment:
I don’t think we should make this method available to templates, it is too dangerous (doesn’t handle insert tags for example).

The other two methods are fine and should be all we need I think.

Reply:
What's with something like this?

```php
<pre>
<?= $this->raw($this->encodedHtmlExample) ?>
</pre>
```

Reply:
Directly printing to the HTML is not safe and could inject basically anything.
But your example would be unsafe with the other two methods too.